### DISCUSSION TITLE
Unable to use Fieldsets in Block

### DISCUSSION THREAD

**Discussion by Tom-revere** [area17/twill#2590] [Ideas]
## Description

I am trying to create Fieldsets within a block using Form Builder, however they never render in the CMS.

### Steps to reproduce

Create a new block
Replace the `getForm` function with the below
```
public function getForm(): Form
{        
    $form = Form::make();

    $form->add(Input::make()->name('title'));
    
    $form->withFieldSets(new Fieldsets([
        Fieldset::make()->title('Fieldset 1')->id('fieldset1')->fields([
            Input::make()
            ->name('fs1_content')
            ->label(twillTrans('FS 1 Content'))
        ]),
        Fieldset::make()->title('Fieldset 2')->id('fieldset2')->fields([
            Wysiwyg::make()
                ->name('fs2_content')
                ->label(twillTrans('FS 2 Content'))
            
        ])
    ]));

    $form->add(Input::make()->name('description'));

    return $form;
}
```

### Expected result

It should show Fieldsets between Title and Description.

### Actual result

It does not show any fieldsets, or the fields within these. Only Title and Description show.

### Versions

Twill 3.2.0
Laravel 9.5.2
PHP 8.0.11
MySQL 5.7.33

**Comment by ifox** [comment]
Fieldsets are not currently supported in blocks and they haven't been designed to be supported there. 

Moving this to a discussion, I'm curious to hear more about the use case for fieldsets in blocks.

**Comment by Tom-revere** [comment]
I personally use this for keeping blocks more concise. I want to give the editors a block which just works without much changes, but I also want to give them the flexibility of changing some of the smaller stuff (such as margins/paddings). With these options I generally put them in an "Advanced" fieldset so they can be changed if needed, but can also just be left well alone.

If these sat outside of a fieldset, it can make the edit page quite long and difficult to navigate (imagine having 10-20 blocks on a page each with 4-6 fields that might never need to be changed).

But it can also be good to separate out the components of a block. So for example having a block which has a Left column, and a Right Column. Using the built-in Columns in the Form Builder can make it seem quite confined when editing (an example below). Allowing these fields to be placed in a fieldset can make this a bit more visually appealing.

![image](https://github.com/area17/twill/assets/169990447/b8c39471-4e07-4daa-adfc-fc5d6c2b4915)
